### PR TITLE
Update with upp/10.0.9 and turn on inline post rt tests in debug mode.

### LINKED ISSUE #686: WRITE_DOPOST error when running in debug mode
## Description
Current control_debug regression test specifies the parameter WRITE_DOPOST=.false. If this value is changed to .true., an error results during run time. However, this error does not occur if gnu compiler is used.

### To Reproduce:
On Hera
1. Edit tests/tests/control_debug to include `export WRITE_DOPOST=.true.`
2. ./rt.sh -n control_debug >out 2>&1
3. This will result in error during the run
4. export RT_COMPILER=gnu
5. ./rt.sh -n control_debug >out 2>&1
6. This will run successfully

## Output
```
144:  post_fname=GFSPRS.GrbF00
  0:  in fcst run phase 2, na=           1
145: forrtl: error (65): floating invalid
145: Image              PC                Routine            Line        Source     
145: fv3.exe            000000000A1F819E  Unknown               Unknown  Unknown
145: libpthread-2.17.s  00002B3AD6C8D630  Unknown               Unknown  Unknown
145: fv3.exe            0000000009DF1AD2  bndlyr_                   311  BNDLYR.f
145: libiomp5.so        00002B3AD539DA43  __kmp_invoke_micr     Unknown  Unknown
145: libiomp5.so        00002B3AD53612C6  __kmp_fork_call       Unknown  Unknown
145: libiomp5.so        00002B3AD5320BB0  __kmpc_fork_call      Unknown  Unknown
145: fv3.exe            0000000009DF70CE  bndlyr_                   272  BNDLYR.f
145: fv3.exe            0000000009D5C9D9  miscln_                  1712  MISCLN.f
145: fv3.exe            0000000009AAA697  process_                  116  PROCESS.f
145: fv3.exe            0000000003169875  post_gfs_mp_post_         201  post_gfs.F90
145: fv3.exe            000000000316512E  inline_post_mp_in          43  inline_post.F90
145: fv3.exe            000000000310333A  module_wrt_grid_c        1489  module_wrt_grid_comp.F90
```

### PR DESCRIPTION
**Pull request by WenMeng-NOAA**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

- Update upp lib to 10.0.9.
- Update UPP control file for RRFS inline post.
- Turn on inline post in control_debug test.
- Add regional_quilt_debug test which includes inline post.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes #686 

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [x] CI: https://github.com/ufs-community/ufs-weather-model/actions/runs/1214185065

## Dependencies

None

### CODE REVIEW

File: tests/tests/control_debug [hunk 1]
```diff
@@ -25,6 +25,7 @@ export OUTPUT_GRID='gaussian_grid'
 export NSTF_NAME='2,0,0,0,0'
 export IAER=5111
 export FHMAX=1
+export WRITE_DOPOST=.true.
```
Comment (DusanJovic-NOAA) [inline tests/tests/control_debug]: If this test now runs inline post, shouldn't we compare post grib files against the baselines?

Comment (WenMeng-NOAA) [reply tests/tests/control_debug]: @DusanJovic-NOAA and @junwang-noaa Please advise the changes of adding grib2 file comparison.

**Comment by junwang-noaa** [comment]
Wen, please add the POST files comparison in tests/tests/control_debug
<https://github.com/ufs-community/ufs-weather-model/pull/773#discussion_r704443613>
:

 export LIST_FILES="sfcf000.nc \
                    sfcf001.nc \
                    atmf000.nc \
-                   atmf001.nc"
+                   atmf001.nc \
+                   GFSFLX.GrbF00 \
+                   GFSFLX.GrbF01 \
+                   GFSPRS.GrbF00 \
+                   GFSPRS.GrbF01"

**Comment by WenMeng-NOAA** [comment]
@junwang-noaa Added the inline post comparison in control_debug. Thanks!

**Review by climbfuji — APPROVED** [review]

**Comment by MinsukJi-NOAA** [comment]
@WenMeng-NOAA can you please make changes in two files for CI container library update to upp 10.0.9?
```
diff --git a/tests/ci/Dockerfile b/tests/ci/Dockerfile
index 34feec3..3534560 100644
--- a/tests/ci/Dockerfile
+++ b/tests/ci/Dockerfile
@@ -1,4 +1,4 @@
-From noaaemc/ubuntu-hpc:v1.6b
+From noaaemc/ubuntu-hpc:v1.7b
 
 ENV HOME=/home/builder
 COPY --chown=builder:builder . $HOME/ufs-weather-model

diff --git a/tests/ci/ci.sh b/tests/ci/ci.sh
index 7702ed3..05f0bed 100755
--- a/tests/ci/ci.sh
+++ b/tests/ci/ci.sh
@@ -81,7 +81,7 @@ elif [ $RUN == "true" ]; then
                 -e "RT_MACHINE=linux.gnu" -e "RT_COMPILER=gnu" \
                 -w "/home/builder/ufs-weather-model/tests" \
                 -v DataVolume:/home/builder/data/NEMSfv3gfs/input-data-20210528 \
-                --shm-size=512m --name my-container noaaemc/ubuntu-hpc:v1.6b \
+                --shm-size=512m --name my-container noaaemc/ubuntu-hpc:v1.7b \
                 /bin/bash -c "./utest -n ${TEST_NAME} -c ${TEST_CASE} -x"
 
   cd $GITHUB_WORKSPACE
```

**Comment by WenMeng-NOAA** [comment]
@MinsukJi-NOAA The changes for CI container were committed.  Thanks!

**Review by MinsukJi-NOAA — APPROVED** [review]

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: BL
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/720601014/20210908134510/ufs-weather-model
Please manually delete: /glade/scratch/dtcufsrt/FV3_RT/rt_47552
Test control_lseaspray 021 failed failed 
Test control_lseaspray 021 failed in run_test failed 
Please make changes and add the following label back:
cheyenne-intel-BL

**Comment by DeniseWorthen** [comment]
The cheyenne.intel error is very odd, looks like it was internal to ESMF?

```
79:MPT: #14 0x0000000000bdd4cf in esmf_iogridmosaicmod::esmf_gridspecreadstaggerr8 (
79:MPT:     filename=<error reading variable: value requires 88297584 bytes, which is more than max-value-size>,
79:MPT:     nx=<error reading variable: Cannot access memory at address 0xffffffff>,
79:MPT:     ny=0, lon=<error reading variable: Location address is not set.>,
79:MPT:     lat=<error reading variable: Location address is not set.>,
79:MPT:     staggerloc=..., start=..., count=..., rc=0, .tmp.FILENAME.len_V$104f=23)
79:MPT:     at /glade/p/ral/jntp/GMTB/tools/hpc-stack-v1.2.0/src/hpc-stack-intel-2021.2/pkg/ESMF_8_2_0_beta_snapshot_14/src/Infrastructure/IO/interface/ESMF_IOGridmosaic.F90:950
```

**Comment by climbfuji** [comment]
> The cheyenne.intel error is very odd, looks like it was internal to ESMF?
> 
> ```
> 79:MPT: #14 0x0000000000bdd4cf in esmf_iogridmosaicmod::esmf_gridspecreadstaggerr8 (
> 79:MPT:     filename=<error reading variable: value requires 88297584 bytes, which is more than max-value-size>,
> 79:MPT:     nx=<error reading variable: Cannot access memory at address 0xffffffff>,
> 79:MPT:     ny=0, lon=<error reading variable: Location address is not set.>,
> 79:MPT:     lat=<error reading variable: Location address is not set.>,
> 79:MPT:     staggerloc=..., start=..., count=..., rc=0, .tmp.FILENAME.len_V$104f=23)
> 79:MPT:     at /glade/p/ral/jntp/GMTB/tools/hpc-stack-v1.2.0/src/hpc-stack-intel-2021.2/pkg/ESMF_8_2_0_beta_snapshot_14/src/Infrastructure/IO/interface/ESMF_IOGridmosaic.F90:950
> ```

Is it repeatable?

**Comment by DeniseWorthen** [comment]
I was just about to give it a try. I'll let you know.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: BL
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/720601014/20210908193005/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_10337
Test control_debug 045 failed failed 
Test control_debug 045 failed in run_test failed 
Test regional_quilt_debug 064 failed failed 
Test regional_quilt_debug 064 failed in run_test failed 
Test compile 002 failed failed 
Test compile 001 failed failed 
Test compile 005 failed failed 
Test control_c384 016 failed failed 
Test control_c384 016 failed in run_test failed 
Please make changes and add the following label back:
gaea-intel-BL

**Comment by DeniseWorthen** [comment]
I was able to create the bl for the failed ``control_lseaspray`` test. I'll move the baseline that was created by Auto-BL and add the seapray test and then set off the Auto-RT.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/720601014/20210908194517/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_2017
Baseline creation and move successful
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/720601014/20210908210117/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_18680
Test compile 002 failed failed 
Please make changes and add the following label back:
hera-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: BL
Repo location: /work/noaa/nems/emc.nemspara/autort/pr/720601014/20210908143012/ufs-weather-model
Please manually delete: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_433775
Please make changes and add the following label back:
orion-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: BL
Repo location: /work/noaa/nems/emc.nemspara/autort/pr/720601014/20210910084523/ufs-weather-model
/work/noaa/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20210907/INTEL
 Exists already. It should not yet. Please delete.
Please make changes and add the following label back:
orion-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: RT
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/720601014/20210910074509/ufs-weather-model
Please manually delete: /glade/scratch/dtcufsrt/FV3_RT/rt_4087
Test regional_quilt_debug 077 failed in check_result failed 
Test regional_quilt_debug 077 failed in run_test failed 
Please make changes and add the following label back:
cheyenne-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: RT
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/720601014/20210910134513/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_32741
Test regional_quilt_debug 078 failed in check_result failed 
Test regional_quilt_debug 078 failed in run_test failed 
Please make changes and add the following label back:
hera-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: RT
Repo location: /work/noaa/nems/emc.nemspara/autort/pr/720601014/20210910090017/ufs-weather-model
Please manually delete: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_389596
Test regional_quilt_debug 078 failed in check_result failed 
Test regional_quilt_debug 078 failed in run_test failed 
Please make changes and add the following label back:
orion-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: BL
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/720601014/20210910134506/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_15930
Test cpld_bmark_v16 005 failed failed 
Test cpld_bmark_v16 005 failed in run_test failed 
Test cpld_bmark_wave_v16_p7b 008 failed failed 
Test cpld_bmark_wave_v16_p7b 008 failed in run_test failed 
Test regional_quilt_RRTMGP 037 failed failed 
Test regional_quilt_RRTMGP 037 failed in run_test failed 
Please make changes and add the following label back:
gaea-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: RT
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/720601014/20210910134516/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_140733
Test regional_quilt_debug 076 failed in check_result failed 
Test regional_quilt_debug 076 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: RT
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/720601014/20210911000036/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_16922
Test control_csawmg 036 failed failed 
Test control_csawmg 036 failed in run_test failed 
Please make changes and add the following label back:
hera-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: RT
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/720601014/20210912033005/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_21140
Test cpld_bmark_wave_v16 013 failed failed 
Test cpld_bmark_wave_v16 013 failed in run_test failed 
Test cpld_bmark_v16_nsst 012 failed failed 
Test cpld_bmark_v16_nsst 012 failed in run_test failed 
Test hafs_regional_atm 084 failed failed 
Test hafs_regional_atm 084 failed in run_test failed 
Please make changes and add the following label back:
gaea-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: BL
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/720601014/20210913054510/ufs-weather-model
/glade/p/ral/jntp/GMTB/ufs-weather-model/RT/NEMSfv3gfs/develop-20210907/INTEL
 Exists already. It should not yet. Please delete.
Please make changes and add the following label back:
cheyenne-intel-BL

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**